### ISSUE TITLE
Accessibility problems

### ISSUE THREAD

**Issue by hhpmusic** [RickStrahl/MarkdownMonster#470]
Hello,
  I'm a blind musician, and just downloaded the Markdown Monster. I'm using Markdown for a documentation I'm writing, and this tool is very useful. But I discover a problem: many buttons in the software, e.g., the ones in the addins manager, are not read by screen readers. The buttons are lack of text labels, and screen readers only read "button" when tabbing on them. The second problem is the list of addins manager can't be read, only "addin items' are read, not their names. The last problem is, when I open a md file, it can be navigated using screen readers; but when I quit the program and open it again, the document is launched but can't be navigated by screen reader. I don't know what's the problem. Could you please solve the above accessibility issues? You can use the free NVDA screen reader to test the accessibility. Thank you very much!

Regards
Haipeng

**Comment by hhpmusic** [comment]
I tried it again, and the biggest problem is at the addin list, where
addins are not read. Also, please take a look at all buttons in your
dialogs, to see whether all of them have text labels.

Haipeng


On 1/5/19, Rick Strahl <notifications@github.com> wrote:
> Thanks for bringing this to my attention. To be honest I haven't given a lot
> of thought to this, but I'll take a look.
>
> Specific areas that are problematic would be helpful.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/RickStrahl/MarkdownMonster/issues/470#issuecomment-451543098


-- 
Hu Haipeng
Blind musician and braille music transcriber of China
THe BrailleOrch and Open Braille Music Projects
http://www.brailleorch.org

**Comment by RickStrahl** [comment]
I added some small adjustments and the addin list should give you at least a list of the items (not sure about the detail - I'm using Narrator for reference at the moment).  

You need to use one of the preview builds (1.14.5.3 or later) which has that change from the [download page](https://markdownmonster.west-wind.com/download.aspx) - click on the preview release link after the main download button. It'll also be in the next update.

That said, I think there are going to be quite a few places that don't work properly for accessibility. Any combo and list boxes, any custom lists (like the Addin list) etc. are not working by default so they'll need work. 

 I'll go through this at some point, but it'll probably not all happen at once and it's best to report each area individually for fixing if possible.

Again thanks for bringing this up - it's not something I had considered previously.

**Comment by hhpmusic** [comment]
Thank you very much, the list of addins is now readable. You can try
all buttons and other objects in your program to see if they are all
readable. Generally, if Narrator can read them correctly, then other
screen readers will be ok.

Haipeng

**Comment by RickStrahl** [comment]
Closing due to inactivity. This topic should be re-opened when there's additional specific feedback or if somebody wants to help out in improving accessibility in general.
